lib/exercism/participation_stats.rb

```diff
@@ -0,0 +1,99 @@
+require 'date'
+
+class ParticipationStats
+  PRE_GAMIFICATION_COMPARISON_DATE = '2017-03-20'
+  GAMIFICATION_START_DATE          = '2017-03-27'
+  GAMIFICATION_WITHDRAWAL_DATE     = '2017-04-10'
+  GAMIFICATION_EXPERIMENT_END_DATE = '2017-04-24'
+
+  attr_reader :date_range, :gamification_markers, :experiment_group
+
+  def self.experiment_complete?
+    Time.now > Date.parse(GAMIFICATION_EXPERIMENT_END_DATE) + 1.day
+  end
+
+  def initialize(date_range, gamification_markers: false, experiment_group: nil)
+    @date_range = date_range
+    @gamification_markers = gamification_markers
+    @experiment_group = experiment_group
+  end
+
+  def results
+    result = {
```

Comment:
The order is important. The trick is that these values get converted immediately to JSON for storage in a data attribute for the Plotly, and this is the format that it expects. In other circumstances, I agree that would make more sense.

See where this is used in [stats.coffee:40](https://github.com/exercism/exercism.io/pull/3447/files/8ee996dd3cc3fc183eca9ac3cb2931a11648633c#diff-6205521a3a951356d9b0411c57b86b6dR40).